--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -22,10 +22,10 @@ noinst_HEADERS = iz.h
 izcoordinator_SOURCES = coordinator.c
 izcoordinator_CFLAGS = $(AM_CFLAGS) $(NL_CFLAGS) -DLEASE_FILE=\"$(leasefile)\"
 izcoordinator_CFLAGS += -DPID_FILE=\"$(pidfile)\"
-izcoordinator_LDADD = ../addrdb/libaddrdb.la $(LDADD) $(NL_LIBS) $(LEXLIB)
+izcoordinator_LDADD = ../addrdb/libaddrdb.la $(LDADD) $(NL_LIBS) -lm $(LEXLIB)
 
 iz_CFLAGS = $(AM_CFLAGS) $(NL_CFLAGS)
-iz_LDADD = $(LDADD) $(NL_LIBS)
+iz_LDADD = $(LDADD) $(NL_LIBS) -lm
 
 izattach.8: $(izattach_SOURCES) $(top_srcdir)/configure.ac
 	-$(HELP2MAN) -o $@ -s 8 -N -n $(izattach_DESC) $(builddir)/izattach
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -277,9 +277,9 @@ noinst_HEADERS = iz.h
 izcoordinator_SOURCES = coordinator.c
 izcoordinator_CFLAGS = $(AM_CFLAGS) $(NL_CFLAGS) \
 	-DLEASE_FILE=\"$(leasefile)\" -DPID_FILE=\"$(pidfile)\"
-izcoordinator_LDADD = ../addrdb/libaddrdb.la $(LDADD) $(NL_LIBS) $(LEXLIB)
+izcoordinator_LDADD = ../addrdb/libaddrdb.la $(LDADD) $(NL_LIBS) -lm $(LEXLIB)
 iz_CFLAGS = $(AM_CFLAGS) $(NL_CFLAGS)
-iz_LDADD = $(LDADD) $(NL_LIBS)
+iz_LDADD = $(LDADD) $(NL_LIBS) -lm
 all: all-am
 
 .SUFFIXES:
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -5,5 +5,5 @@ zbtest_PROGRAMS = listen-packet test tes
 izlisten_SOURCES = listen.c
 
 gnl_CFLAGS = $(AM_CFLAGS) $(NL_CFLAGS)
-gnl_LDADD = $(LDADD) $(NL_LIBS)
+gnl_LDADD = $(LDADD) $(NL_LIBS) -lm
 
--- a/tests/Makefile.in
+++ b/tests/Makefile.in
@@ -259,7 +259,7 @@ AM_CFLAGS = -g3 -Wall
 LDADD = $(top_builddir)/lib/libcommon.la
 izlisten_SOURCES = listen.c
 gnl_CFLAGS = $(AM_CFLAGS) $(NL_CFLAGS)
-gnl_LDADD = $(LDADD) $(NL_LIBS)
+gnl_LDADD = $(LDADD) $(NL_LIBS) -lm
 all: all-am
 
 .SUFFIXES:
